src/ws2st/refactor.cpp

```diff
@@ -10,159 +10,152 @@
+            WashRefactoringAction(&variables::GetVariableMatcher, &variables::HandleGetVariable),
+            // Replacement to add the force definitions
+            WashRefactoringAction(&forces::InsertForcesDefinitionMatcher, &forces::HandleInsertForcesDefinition),
+
+            WashRefactoringAction(&meta::SimulationVecTMatcher, &meta::HandleSimulationVecTMatcher)
+        },
+
+        // 2nd pass: set calls
+        {
+            // Calls to set a force
+            WashRefactoringAction(&forces::SetForceScalarMatcher, forces::HandleSetForceScalar),
+            WashRefactoringAction(&forces::SetForceVectorMatcher, forces::HandleSetForceVector),
+            // Calls to set pred-defined particle properties
+            WashRefactoringAction(&forces::SetPosMatcher, forces::HandleSetPos),
+            WashRefactoringAction(&forces::SetVelMatcher, forces::HandleSetVel),
+            WashRefactoringAction(&forces::SetAccMatcher, forces::HandleSetAcc),
+        
+            WashRefactoringAction(&forces::SetDensityMatcher, forces::HandleSetDensity),
+            WashRefactoringAction(&forces::SetMassMatcher, forces::HandleSetMass),
+            WashRefactoringAction(&forces::SetSmoothingLengthMatcher, forces::HandleSetSmoothingLength),
+
+            // Calls to set a variable
+            WashRefactoringAction(&variables::SetVariableMatcher, &variables::HandleSetVariable),
+            // WashRefactoringAction(forces::IOForcesLoopMatcher, &forces::HandleIOForcesLoop),
         }
-    }
-
-    int getForceRewriting(RefactoringTool& Tool, const std::unordered_set<std::string>& scalar_f, const std::unordered_set<std::string>& vector_f) {
-        ASTMatchRefactorer finder(Tool.getReplacements());
-        refactoring::GetForceRefactor<refactoring::TypeKind::SCALAR> getForcesScalar;
-        refactoring::GetForceRefactor<refactoring::TypeKind::VECTOR> getForcesVector;
-
-        refactoring::GetParticlePropertyRefactor<refactoring::SCALAR> getMass("mass");
-        refactoring::GetParticlePropertyRefactor<refactoring::SCALAR> getDensity("density");
-        refactoring::GetParticlePropertyRefactor<refactoring::SCALAR> getSL("smoothing_length");
-
-        refactoring::GetParticlePropertyRefactor<refactoring::VECTOR> getPos("pos");
-        refactoring::GetParticlePropertyRefactor<refactoring::VECTOR> getVel("vel");
-        refactoring::GetParticlePropertyRefactor<refactoring::VECTOR> getAcc("acc");
-
-        refactoring::AddForcDeclarationsRefactor forceDeclarations(scalar_f, vector_f);
 
-        finder.addMatcher(forceArrays, &forceDeclarations);
+    };
 
-        finder.addMatcher(getForceScalarMatcher, &getForcesScalar);
-        finder.addMatcher(getForceVectorMatcher, &getForcesVector);
+    void runRefactoringStages(std::vector<std::string> argv) {
+        int new_argc = 0;
+        std::vector<const char*> args_cstr = {};
+        std::transform(argv.cbegin(), argv.cend(), std::back_inserter(args_cstr),
```

Comment:
No idea how this works but looks pretty cool

Reply:
A clang refactor tool has a set of matchers and callback classes. The `WashRefactoringAction` class is essentially just a pair of any <T>Matcher type and the callback function type. Then when the tool starts we can run through each list in the 2d list and create the callback class and add it to the tool. Previously did it manually but it's a lot more code to write the whole class boilerplate and add it to the tool that way so developed this to make it easier.